Script to render tree instance IDs

amrithasp02 opened the pull request.

amrithasp02 commented:
On an outdated branch before the config changes the script seems to run without issues and saves the rendered IDs to `/ofo-share/species-prediction-project/intermediate/rendered_instance_ids`. However, when I test this script in this branch having the latest changes from main it doesn't seem to read the folders that exist (e.g. 0002_000451_000446). 

```
(geograypher-amri) exouser@amritha-dev-03:/ofo-share/repos-amritha/tree-species-prediction/1_data_prep$ python 12_render_instance_ids.py 
/ofo-share/repos-amritha/conda/envs/geograypher-amri/lib/python3.9/site-packages/pyvista/plotting/plotter.py:149: UserWarning: 
This system does not appear to be running an xserver.
PyVista will likely segfault when rendering.

Try starting a virtual frame buffer with xvfb, or using
  ``pyvista.start_xvfb()``

  warnings.warn(
Skipping 0184_001270_001271 - vector file not found at /ofo-share/species-prediction-project/intermediate/drone_crowns_with_field_attributes/0184_001270_001271.gpkg
Skipping 0066_001468_001469 - vector file not found at /ofo-share/species-prediction-project/intermediate/drone_crowns_with_field_attributes/0066_001468_001469.gpkg
Skipping 0142_000208_000210 - vector file not found at /ofo-share/species-prediction-project/intermediate/drone_crowns_with_field_attributes/0142_000208_000210.gpkg
Skipping 0073_000874_000932 - vector file not found at /ofo-share/species-prediction-project/intermediate/drone_crowns_with_field_attributes/0073_000874_000932.gpkg
Loading mesh
Reading the mesh
skipping dataset 0002_000451_000446 because of missing files. The error was the following:
File (/ofo-share/species-prediction-project/intermediate/photogrammetry/0002_000451_000446/outputs/0002_000451_000446_model.ply) not found
Skipping 0048_000174_000171 - vector file not found at /ofo-share/species-prediction-project/intermediate/drone_crowns_with_field_attributes/0048_000174_000171.gpkg
Skipping 0224_000315_000314 - vector file not found at /ofo-share/species-prediction-
```

russelldj commented on 1_data_prep/12_render_instance_ids.py:
```diff
@@ -0,0 +1,93 @@
+from geograypher.utils.visualization import show_segmentation_labels
+
+# Ensure project root is on sys.path so _bootstrap can be found
+sys.path.append(str(Path(__file__).resolve().parent.parent))
```
Did you run the [symlink_bootstrap](https://github.com/open-forest-observatory/tree-species-prediction/blob/main/symlink_bootstrap.py) file?

amrithasp02 replied:
Ah I missed that, thanks

amrithasp02 commented:
> I think this was due to sloppiness on my part. The photogrammetry products that should have been used were saved to `/ofo-share/species-prediction-project/intermediate/metashape_configs_merged_oblique_nadir` and I just updated the `config.py` option to reflect this. But then Brandon's refactor reverted this change (which is fine, mine was sloppy) and broke it. I've now renamed the data files so I think it should work. Can you try the current version of the code again?

I tried it again now, and it works!

amrithasp02 commented on 1_data_prep/12_render_instance_ids.py:
```diff
@@ -0,0 +1,91 @@
+# Portions of the mesh within this distance of the labels are used for rendering
+MESH_BUFFER_RADIUS_METER = 20
+# Cameras within this radius of the annotations are used for training
+CAMERAS_BUFFER_RADIUS_METERS = 10
```
Is 10 okay for now or should I make it higher?

russelldj replied:
Maybe 25 for now?

russelldj commented on 1_data_prep/12_render_instance_ids.py:
```diff
@@ -0,0 +1,97 @@
+VIS = True
+
+RENDER_ROOT_FOLDER = path_config.rendered_instance_ids / "renders"
+VIS_ROOT_FOLDER = path_config.rendered_instance_ids / "visualizations"
```
It might be worth creating a new variable for this so someone doesn't need to recreate this logic in a later script.

amrithasp02 commented on 1_data_prep/12_render_instance_ids.py:
```diff
@@ -0,0 +1,97 @@
+                DTM_file=dtm_file,
+                ground_height_threshold=2,
+                cameras_ROI_buffer_radius_meters=20,
+                mesh_ROI_buffer_radius_meters=20,
```
Done. Also I noticed that come of the variables like `RENDER_IMAGE_SCALE` which were already defined at the top ([see this commit](https://github.com/open-forest-observatory/tree-species-prediction/pull/27/commits/a1b79a7a81f8793fac4aca9baf5afb653472bffb)) were not being passed to `render_labels()` so I also included those. Can you verify if the values look good before I save the renders for all the datasets?
